astropy/coordinates/builtin_frames/ecliptic_transforms.py

```diff
@@ -26,8 +26,19 @@ def _ecliptic_rotation_matrix(equinox):
     jd1, jd2 = get_jd12(equinox, 'tt')
     rnpb = erfa.pnm06a(jd1, jd2)
     obl = erfa.obl06(jd1, jd2)*u.radian
-    return np.asarray(np.dot(rotation_matrix(obl, 'x'), rnpb))
-
+    """
+    The following code is the equivalent of looping over obl and rnpb,
+    creating a rotation matrix from obl and then taking
+    the dot product of the resulting matrices, finally combining
+    into a new array.
+    """
+    try:
+        rmat = np.array([rotation_matrix(this_obl, 'x') for this_obl in obl])
```

Comment:
I was about to write that this seemed to be something `rotation_matrix` should be able to handle, when I realised it could not because it returns `np.matrix`. So, not for this PR, but see #5088.